pyproject.toml

```diff
@@ -68,11 +68,44 @@ pytest-cov = "*"
 sphinx = "^5.3"
 tomli = "*"  # for tests only
 
-[tool.isort]
-profile = "black"
+[tool.ruff]
+target-version = "py310"
+
+[tool.ruff.lint]
+ignore = [
+    "B028",  # no-explicit-stacklevel for warnings.warn()
+    "E501",  # line-too-long - code is reformatted (do not care about comments and docstring)
+    "F401",  # unsused-import - covered by pycln
+    "RUF005", # TO DO
+    "RUF012", # TO DO
+    "UP038", # TO DO
+    "C408", # TO DO
```

Comment:
**issue:** restrict ruff execution on files added by Mira, hopefully can restore these checks

that check only devtools, omf/fileio/geoh5 and tests

Reply:
although other checks are good on all files... Before, only flake8 and pylint were configured to run on a reduced set of files.
Is there a way to tell ruff what files to ignore per check? Or what checks to run per file pattern?

Reply:
Yes, there is a way since 2023 with [per-file-ignores](https://docs.astral.sh/ruff/settings/#lint_per-file-ignores) function.

```
[tool.ruff.lint.per-file-ignores]
# Ignore `E402` (import violations) in all `__init__.py` files, and in `path/to/file.py`.
"__init__.py" = ["E402"]
"path/to/file.py" = ["E402"]
# Ignore `D` rules everywhere except for the `src/` directory.
"!src/**.py" = ["D"]
```